R/exclude.R

```diff
@@ -65,27 +87,85 @@ parse_exclusions <- function(file, exclude = settings$exclude,
     }
 
     for (i in seq_along(starts)) {
-      exclusions <- c(exclusions, seq(starts[i], ends[i]))
+      excluded_lines <- seq(starts[i], ends[i])
+      linters_string <- substring(lines[starts[i]], start_locations[starts[i]])
+      linters_string <- rex::re_matches(linters_string, exclude_linter)[, 1L]
+
+      # No match for linter list: Add to global excludes
+      if (is.na(linters_string)) {
+        exclusions <- add_excluded_lines(exclusions, excluded_lines, "")
+      } else {
+        # Matched a linter list: only add excluded lines for the listed linters.
+        excluded_linters <- strsplit(linters_string, exclude_linter_sep)[[1L]]
+        exclusions <- add_excluded_lines(exclusions, excluded_lines, excluded_linters)
+      }
     }
   }
 
-  sort(unique(c(exclusions, which(rex::re_matches(lines, exclude)))))
+  nolint_locations <- rex::re_matches(lines, exclude, locations = TRUE)[, "end"] + 1L
+  nolints <- which(!is.na(nolint_locations))
+  # Disregard nolint tags if they als match nolint start / end
+  nolints <- setdiff(nolints, c(starts, ends))
+
+  for (i in seq_along(nolints)) {
+    linters_string <- substring(lines[nolints[i]], nolint_locations[nolints[i]])
+    linters_string <- rex::re_matches(linters_string, exclude_linter)[, 1L]
+
+    # No match for linter list: Add to global excludes
+    if (is.na(linters_string)) {
```

Comment:
this logic is repeated for the range & one-line case, could you put it to a helper function?

Reply:
i guess this is a place where python's unpack-and-assign would come in handy for writing the helper...

Reply:
Done.